Documentation/Evolution/RegexSyntax.md

```diff
@@ -619,12 +619,12 @@ OnigurumaCalloutArgList -> OnigurumaCalloutArg (',' OnigurumaCalloutArgList)*
 OnigurumaCalloutArg     -> [^,}]+
 OnigurumaTag            -> '[' Identifier ']'
 
-OnigurumaCalloutOfContents -> '(?' '{'+ Contents '}'+ OnigurumaTag? Direction? ')'
-OnigurumaCalloutContents   -> <String>
+OnigurumaCalloutOfContents -> '(?' '{' OnigurumaCalloutContents '}' OnigurumaTag? Direction? ')'
+OnigurumaCalloutContents   -> <String> | '{' OnigurumaCalloutContents '}'
 OnigurumaCalloutDirection  -> 'X' | '<' | '>'
 ```
 
-A callout is a feature that allows a user-supplied function to be called when matching reaches that point in the pattern. We supported parsing both the PCRE and Oniguruma callout syntax. The PCRE syntax accepts a string or numeric argument that is passed to the function. The Oniguruma syntax is more involved, and may accept a tag, argument list, or even an arbitrary program in the 'callout of contents' syntax.
+A callout is a feature that allows a user-supplied function to be called when matching reaches that point in the pattern. We supported parsing both the PCRE and Oniguruma callout syntax. The PCRE syntax accepts a string or numeric argument that is passed to the function. The Oniguruma syntax is more involved, and may accept an identifier with an optional tag and argument list. It may also accept an arbitrary program in the 'callout of contents' syntax. This is an expanded version of Perl's interpolation syntax, and allows an arbitrary nesting of delimiters in addition to an optional tag and direction.
```

Comment:
Worth an additional paragraph perhaps:

> While we propose parsing these for the purposes of issuing helpful diagnostics, we are deferring full support for interpolation for the future.

We may want to remove the "Oniguruma" from the title of some of the productions, as this may end up being the Swift interpolation syntax. We just need to figure out what types of things go inside there (analogous to string interpolations) and if that's protocol driven. Clearly future work, though it would bring literals more up to parity with result builders' ability to interweave library calls.

Reply:
How does this look?